### ISSUE TITLE
[Badge] axe violation - a11y error!button-name on 1 Node

### ISSUE THREAD

**Issue by divyajindel** [Sage/carbon#5701]
### Current behaviour

<img width="402" alt="Screenshot 2022-12-20 at 13 49 58" src="https://user-images.githubusercontent.com/88449494/208682371-9f99a08a-d611-4ff4-8e34-a01544a852ce.png">

### Expected behaviour

Axe Violation should not occur.

### CodeSandbox or Storybook URL

https://carbon.sage.com/?path=/docs/badge--default-story

### Carbon Version

latest

### What browsers are you seeing the problem on?

Chrome

### What Operating System are you seeing the problem on?

MacOS

### Anything else we should know?

No Violation occurred for FireFox.

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by tomdavies73** [comment]
FE-5596
